src/Infrastructure/Service/LifecycleManager.php

```diff
@@ -51,4 +78,79 @@ public function update(): void
             $action->apply($taskIds, $folderTasks, $databaseTasks);
         }
     }
+
+    /**
+     * @throws \SprykerSdk\Sdk\Infrastructure\Exception\SdkVersionNotFoundException
+     *
+     * @return array<\SprykerSdk\SdkContracts\Entity\MessageInterface>
+     */
+    public function checkForUpdate(): array
+    {
+        $versionFilePath = $this->sdkDirectory . '/VERSION';
+
+        if (!file_exists($versionFilePath)) {
+            throw new SdkVersionNotFoundException('Could not find VERSION file, skip updatable check');
+        }
+
+        $currentVersion = file_get_contents($versionFilePath);
+
+        if (!$currentVersion) {
+            throw new SdkVersionNotFoundException('Could not find VERSION file, skip updatable check');
+        }
+
+        $messages = [];
+        $currentVersion = trim($currentVersion);
+
+        try {
+            $latestVersion = $this->getLatestVersion();
+        } catch (SdkVersionNotFoundException $exception) {
+            $latestVersion = static::GITHUB_VERSION;
+            $messages[] = new Message($exception->getMessage());
+        }
+
+        if (version_compare($currentVersion, $latestVersion, '<')) {
+            $messages[] = new Message(sprintf('SDK is outdated (current: %s, latest: %s)', $currentVersion, $latestVersion));
+            $messages[] = new Message('Please update manually by downloading the installer for the newest version at https://github.com/spryker-sdk/sdk/releases');
+        }
+
+        return $messages;
+    }
+
+    /**
+     * @throws \SprykerSdk\Sdk\Infrastructure\Exception\SdkVersionNotFoundException
+     *
+     * @return string
+     */
+    protected function getLatestVersion(): string
+    {
+        $opts = [
+            'http' => [
+                'method' => 'GET',
+                'header' => [
+                    'User-Agent: PHP',
+                ],
+                'timeout' => 10,
+            ],
+        ];
+
+        $context = stream_context_create($opts);
+
+        try {
+            $content = file_get_contents(static::GITHUB_ENDPOINT, false, $context);
+
+            if (!$content) {
+                throw new SdkVersionNotFoundException(sprintf('Could not read from %s', static::GITHUB_ENDPOINT));
```

Comment:
Would be fine to get more verbose message.
```
$content = @file_get_contents(static::GITHUB_ENDPOINT, false, $context);

if ($content === false) {
   throw new SdkVersionNotFoundException(sprintf('Could not read from %s, error: %s', static::GITHUB_ENDPOINT, error_get_last()['message'] ?? ''));
}
```

Reply:
Fixed